lib/serial_terminal.pm

```diff
@@ -59,20 +59,39 @@ sub add_serial_console {
     prepare_serial_console();
 
 Wrapper for add_serial_console.
+
+Configure serial consoles for virtio support (root-virtio-terminal and
+user-virtio-terminal).
+
+NOTE: if test plans to use more consoles via VIRTIO_CONSOLE_NUM, it have to
+call add_serial_console() with proper console name (beware different number
+for ppc64le).
 =cut
 
 sub prepare_serial_console {
-    # Configure serial consoles for virtio support
-    # poo#18860 Enable console on hvc0 on SLES < 12-SP2
-    # poo#44699 Enable console on hvc1 to fix login issues on ppc64le
+    script_run('systemctl | grep serial-getty');
+    record_info('getty before', script_output('systemctl | grep serial-getty'));
+
     if (!check_var('VIRTIO_CONSOLE', 0)) {
+        my $console = 'hvc1';
+
+        # poo#18860 Enable console on hvc0 on SLES < 12-SP2 (root-virtio-terminal)
```

Comment:
```
        # poo#18860 Enable console on hvc0 on SLES < 12-SP2 (root-virtio-terminal)
        if (is_sle('<12-SP2') && !is_s390x) {
            add_serial_console('hvc0');
        }
        # poo#44699 Enable console on hvc1 to fix login issues on ppc64le
        # (root-virtio-terminal)
        elsif (get_var('OFW')) {
            add_serial_console('hvc1');
            add_serial_console('hvc2');
        } else {
             # user-virtio-terminal
             add_serial_console('hvc1');
         }
```

IMHO  beside one line duplication I find this version more readable and clear for people who will read this later  and will try to understand what is going on here

Reply:
This way `is_sle('<12-SP2') && !is_s390x` will not get extra console for `user-virtio-terminal`. This must be always added. Once we delete whole `if (is_sle('<12-SP2') && !is_s390x) {`, but not in this effort.

Reply:
not sure about others but for me this variable in your version really prevent me from understanding what is going on in the code

Reply:
> not sure about others but for me this variable in your version really prevent me from understanding what is going on in the code

@asmorodskyi Code does the same, but in your way the comments are wrong. `# user-virtio-terminal` means that this console is for user console, but actually in ppc64le is used for root console. Also `# poo#44699 Enable console on hvc1 to fix login issues on ppc64le (root-virtio-terminal)` is wrong, because `hvc2` is on ppc64le for actually used for user virtio terminal (but user terminal for other archs; for them is comment valid).

The logic in my code is: `if` and `elseif` is for special cases for root terminal (not sure why older sles needs `hvc0` and never `hvc1`, but it works like this). And `add_serial_console($console);` at the end is for adding user console.

I considered `$user_console` too long, thus I used just `$console`, because `user-virtio-terminal` comment looked to me obvious what is being added.

Anyway, feel free to suggest something else, just comments should be correct and code not too verbose.